pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -35,7 +36,14 @@ func Add(mgr manager.Manager) error {
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileIstioControlPlane{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+	factory := &helmreconciler.Factory{
+		CustomizerFactory: &helmreconciler.CustomizerFactory{
+			InputFactory:    &IstioInputFactory{},
+			MarkingsFactory: &IstioMarkingsFactory{},
+			ListenerFactory: &IstioListenerFactory{},
+		},
```

Comment:
The factory terminology here is a bit confusing. Why do we need to pass in factories?

Reply:
yes, i suppose this could be a single factory with three methods.  i started out by defining the interfaces for input, markings, and hooks (which became a listener/callback interface) and kept the separation through the factories.  i'll look at consolidating these into a single factory interface.